.github/workflows/pypi.yaml

```diff
@@ -0,0 +1,42 @@
+name: Push to Pypi
+
+on:
+  release:
+    types: [released, prereleased]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        pip install --no-cache-dir -e .[complete]
+        pip list
+    - name: Build the func_adl wheel
+      env:
+        func_adl_version: ${{ github.ref }}
+      run: |
+        python setup_func_adl.py sdist bdist_wheel
```

Comment:
Never run python on `setup.py` directly, always use `pip` or `pep517.build`, they allow setuptools to do extra hacks that it can't do otherwise. And why isn't the file called setup.py?

Reply:
It isn't called `setup.py` because this repo builds two packages, one called func_adl, and the other func_adl.ast.